qu19041_ex3.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import matplotlib.pyplot as pyplot
+
+# PART (A)
+
+def integrand(screen_coord, apt_coord, c):
+    return np.exp( c['waveno'] * 1j * (screen_coord - apt_coord) ** 2 / (2 * c['screendist']) )
+
+def fresnel_integral(screen_coord, apt, c):
+    assert c['intervals'] % 2 == 0, "number of intervals must be even."
+
+    aperture = np.linspace(apt['apt_lower'], apt['apt_upper'], c['intervals'] + 1)
```

Comment:
Try and be consistent about leaving spaces after opening parentheses. Sometimes you do, sometimes you  don't. Personally, I don't bother. I prefer to leave a space after the if. Like, if (blah()): The important thing to do is settle on your style and stick to it.